### ISSUE TITLE
Support completion with xs:any

### ISSUE THREAD

**Issue by ehigiepaul** [redhat-developer/vscode-xml#177]
I just noticed that the intellisense is broken in the latest version.
I had to reverse back to version 0.6.0 which works very well and also the validator fails too.

Please kindly review this issue and thanks for this nice extension.

**Comment by fbricon** [comment]
Please provide sample files reproducing the issue.

**Comment by ehigiepaul** [comment]
[sample.zip](https://github.com/redhat-developer/vscode-xml/files/3482737/sample.zip)

Also on the lower versions no validator is working

**Comment by fbricon** [comment]
How did you associate the xsd to the xml? file association setting? catalog?

**Comment by fbricon** [comment]
Ok, I can reproduce with 
```
  "xml.fileAssociations": [
      {
          "pattern": "**/*.xml",
          "systemId": "tns.xsd"
      }
  ]
```

@NikolasKomonen something you could check?

**Comment by fbricon** [comment]
So I get completion under the root (most if not all elements), under the ActionBar element (ActionBar.actionItems), nothing under Page or ScrollView. 

Relates to https://github.com/angelozerr/lsp4xml/issues/180

**Comment by ehigiepaul** [comment]
Just also noticed that validation works on the first tag/root tag but fails on the inner tags.

**Comment by angelozerr** [comment]
@ehigiepaul could you please give us  XML samples which doesn't work with completion and validation because in my case it works:

 * For completion (see cursor position with |)

```xml
<Action | <-- here ActionBar is opened)
```

```xml
<ActionBar>
    | <-- here ActionBar.actionItems is opened
</ActionBar>
```

```xml
<ActionBar>
    <ActionBar.actionItems>
        | <-- here ActionItem is opened
    </ActionBar.actionItems>
</ActionBar>
```

 * for validation it works too:

![image](https://user-images.githubusercontent.com/1932211/62763045-e1ea6980-ba8a-11e9-9803-aed12a513d32.png)

**Comment by angelozerr** [comment]
> nothing under Page or ScrollView.

For elements, yes it's normal (according XML Schema), there are only attributes for thoses elements (completion on attribute should work).

**Comment by ehigiepaul** [comment]
Please kinld reverse to version 0.6.0 and run the xsd and see it all works well. But on version 0.8.0 all fails.

**Comment by angelozerr** [comment]
Could you please give us the case which doesnt work. Have you tested the
completion and validation case that i have explained below.is it working
for you?

Without a detailled use case it is difficult to reproduce your problem

**Comment by ehigiepaul** [comment]
here is the current version 0.8.0 issue. 
- I get attributes on every tag, 
- Validation is present only on the root tag
- No validation on inner tags
- No tag suggestions.
![version 0 8 0 err](https://user-images.githubusercontent.com/8577824/62813380-544a6080-bb02-11e9-88df-6a442327caee.gif)

here is what i get on version 0.6.0. 
- I get attributes on every tag, 
- Validation is present only on the root tag
- No validation on inner tags
- Tag suggestions works
![version 0 6 0](https://user-images.githubusercontent.com/8577824/62813396-7217c580-bb02-11e9-9838-e3466667fee2.gif)

**Comment by skjardenCode** [comment]
Hi,

I have the same problem with XML validation.

VS Code Version: 1.36.1
XML Language Support Version: 0.8.0

XML to validate:

```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<cpr xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="https://www.dgai.de/cpr/schema/ev/cpr-ev-2.0.xsd">
<cprev>
        <VERSION>2.0</VERSION>
        <DATUM>2019-08-09</DATUM>
        <STOKENN>stokenn001</STOKENN>
        <EWZ>00000000</EWZ>
        <PATID>000000000000000</PATID>
        <PROTAR>01</PROTAR>
        <GEBDAT>asd</GEBDAT>
        <PATALTER>999:9999</PATALTER>
        <EOKO>00</EOKO>
        <PACER>06</PACER>
        <FBSYSTEM>06</FBSYSTEM>
        <AUTOCPR>06</AUTOCPR>
        <NTECH>02</NTECH>
        <ROSC>98</ROSC>
        <KHAUF>98</KHAUF>
        <KVEN>999</KVEN>
	<NACHFNA>02</NACHFNA>
    </cprev>
</cpr>
```

There are numerous problems with the above XML:

```
cvc-pattern-valid: Value 'stokenn001' is not facet-valid with respect to pattern '[0-9]{8}' for type 't_STOKENN'.
cvc-type.3.1.3: The value 'stokenn001' of element 'STOKENN' is not valid.
cvc-datatype-valid.1.2.2: 'asd' is not a valid value of list type 't_GEBDAT'.
cvc-type.3.1.3: The value 'asd' of element 'GEBDAT' is not valid.
cvc-datatype-valid.1.2.2: '999:9999' is not a valid value of list type 't_PATALTER'.
cvc-type.3.1.3: The value '999:9999' of element 'PATALTER' is not valid.
cvc-complex-type.2.4.a: Invalid content was found starting with element 'EOKO'. One of '{"":PAEDALT, "":GESCHL}' is expected.
cvc-length-valid: Value '999' with length = '3' is not facet-valid with respect to length '2' for type 't_KVEN'.
cvc-type.3.1.3: The value '999' of element 'KVEN' is not valid.
```

But the extension does not recognize any of those XML validation errors.

**Comment by angelozerr** [comment]
@ehigiepaul thanks for the demo i can reproduce it 

@skjardenCode please create à new issue with your xml validation problem

**Comment by angelozerr** [comment]
@ehigiepaul since 0.8.0 completion show possible elements instead of showing all elements. See `Accurate XSD-based completion` in https://developers.redhat.com/blog/2019/07/24/whats-new-in-the-visual-studio-code-xml-extension/ 

Your XML Schema is using `xs:any` (<xs:any processContents="skip" minOccurs="0" maxOccurs="1" />):

```xml
<xs:complexType name="ContentView">
    <xs:complexContent>
      <xs:extension base="View">
        <xs:sequence>
          <xs:any processContents="skip" minOccurs="0" maxOccurs="1" />
        </xs:sequence>
        <xs:attributeGroup ref="contentViewAttributes" />
      </xs:extension>
    </xs:complexContent>
  </xs:complexType>
```
It is this case with xs:any which is not managed. I'm fixing it.

For validation, it's because you are using `<xs:any processContents="skip"`  which means that validation is skipped. if you change with `<xs:any processContents="lax"` you should have validation.

@skjardenCode there is indeed a bug with your XML usecase, please create à new issue and I will fix it as soon as you will create the issue.

**Comment by angelozerr** [comment]
@skjardenCode please see https://github.com/angelozerr/lsp4xml/issues/553 for your problem which will be fixed by @NikolasKomonen

**Comment by ehigiepaul** [comment]
@angelozerr  thanks for the processContents fix for validation. Works fine.
But still in version 0.8.0 intellisense or tag suggestion issue need to be fixed

**Comment by angelozerr** [comment]
> @angelozerr thanks for the processContents fix for validation. Works fine.

You are welcome!

> But still in version 0.8.0 intellisense or tag suggestion issue need to be fixed

Yes sure, I will work on that. But I think we should show all XML element declared in XML file when xs:any is used. I mean XML element coming from XML Schema and custom XML Element.

For instance, if we have:

```xml
<Page>
  <a />
  <ScrollView>
    | <-- here completion must show Page, Scrollview, ContentView (from XML Schema) and <a /> (custom XML element)
```

**Comment by ehigiepaul** [comment]
@angelozerr  version  report now after your quick xsd fix

This is what i get on the current version 0.8.0 .
attributes on every tag, ✔
Validation on the root tag ✔
validation on inner tags attributes ✔
tag suggestions.❌

here is what i get on version 0.6.0.
attributes on every tag, ✔
Validation on the root tag ✔
validation on inner tags attributes ✔
Tag suggestions works ✔

I

**Comment by ehigiepaul** [comment]
Any hope on the fix

**Comment by angelozerr** [comment]
I'm working on this issue.

**Comment by angelozerr** [comment]
Fixed with https://github.com/angelozerr/lsp4xml/pull/559

**Comment by fbricon** [comment]
Completion returns duplicates from already inserted elements from the xsd:

<img width="828" alt="Screen Shot 2019-08-22 at 6 17 39 PM" src="https://user-images.githubusercontent.com/148698/63531611-281ede80-c509-11e9-9507-d6371115a7da.png">

Using https://github.com/redhat-developer/vscode-xml/files/3482737/sample.zip

**Comment by angelozerr** [comment]
@fbricon good catch. Please review my PR https://github.com/angelozerr/lsp4xml/pull/563 which fixes this duplicate issue.

**Comment by angelozerr** [comment]
> Completion returns duplicates from already inserted elements from the xsd:

Fixed with angelozerr/lsp4xml#563

**Comment by mickaelistria** [comment]
IMO, schema completion in `xs:any` should only be provided when `processContents` is not `skip`. See #612 for example and rationale. The semantic of `processContents` is explicit enough to tell that no schema-related features (including completion) should be involved when set to `skip`.
